libraries/AP_FETtecOneWire/AP_FETtecOneWire.cpp

```diff
@@ -0,0 +1,831 @@
+            }
+            return;
+        case ESCState::WAITING_OK_FOR_START_FW:
+            return;
+#if HAL_AP_FETTEC_ONEWIRE_GET_STATIC_INFO
+        case ESCState::WANT_SEND_REQ_TYPE:
+            if (transmit_config_request(PackedMessage<REQ_TYPE>{esc.id, REQ_TYPE{}})) {
+                esc.set_state(ESCState::WAITING_ESC_TYPE);
+            }
+            return;
+        case ESCState::WAITING_ESC_TYPE:
+            return;
+        case ESCState::WANT_SEND_REQ_SW_VER:
+            if (transmit_config_request(PackedMessage<REQ_SW_VER>{esc.id, REQ_SW_VER{}})) {
+                esc.set_state(ESCState::WAITING_SW_VER);
+            }
+            return;
+        case ESCState::WAITING_SW_VER:
+            return;
+        case ESCState::WANT_SEND_REQ_SN:
+            if (transmit_config_request(PackedMessage<REQ_SN>{esc.id, REQ_SN{}})) {
+                esc.set_state(ESCState::WAITING_SN);
+            }
+            return;
+        case ESCState::WAITING_SN:
+            return;
+#endif
+#if HAL_WITH_ESC_TELEM
+        case ESCState::WANT_SEND_SET_TLM_TYPE:
+            if (transmit_config_request(PackedMessage<SET_TLM_TYPE>{esc.id, SET_TLM_TYPE{1}})) {
+                esc.set_state(ESCState::WAITING_SET_TLM_TYPE_OK);
+            }
+            return;
+        case ESCState::WAITING_SET_TLM_TYPE_OK:
+            return;
+#endif
+        case ESCState::WANT_SEND_SET_FAST_COM_LENGTH: {
+            // FIXME: tidy this up a bit
+            const uint16_t bit_count = 12 + (_esc_count * 11);
+            const uint8_t byte_count = 1 + (bit_count + 7)/8;
+            if (transmit_config_request(PackedMessage<SET_FAST_COM_LENGTH>{esc.id,
+                            SET_FAST_COM_LENGTH{
+                            byte_count,
+                                _escs[0].id,
+                                _esc_count
+                                }})) {
+                esc.set_state(ESCState::WAITING_SET_FAST_COM_LENGTH_OK);
+            }
+        }
+            return;
+        case ESCState::WAITING_SET_FAST_COM_LENGTH_OK:
+            return;
+        case ESCState::RUNNING: {
+            break;
+        }
+        }
+    }
+}
+
+/// periodically called from SRV_Channels::push()
+void AP_FETtecOneWire::update()
+{
+    if (!_init_done) {
+        init();
+        return; // the rest of this function can only run after fully initted
+    }
+
+    // read all data from incoming serial:
+    read_data_from_uart();
+
+    // run ESC configuration state machines
+    if (!hal.util->get_soft_armed()) {
+        configure_escs();
+    }
+
+    // get ESC set points
+    uint16_t motor_pwm[_esc_count];
+    bool some_not_running = false;
+    for (uint8_t i = 0; i < _esc_count; i++) {
+        const ESC &esc = _escs[i];
+        if (esc.state != ESCState::RUNNING) {
+            some_not_running = true;
+        }
+        const SRV_Channel* c = SRV_Channels::srv_channel(esc.id-1);
+        if (c == nullptr) {

```

Comment:
```suggestion
        if (c == nullptr) {
            some_not_running = true;
```

Reply:
I assume this is a cr/lf thing?

I'd actually prefer we change the line endings in these files before merge - there are very few files in our tree which use the line endings in here, and I recall us accepting patches before the canonicalise on a simple line-feed ending.  I can make that change rapidly if this suggestion is accepted - but it would be done once everything was squished to the appropriate commit count.

Reply:
done, as discussed on the discord call